Matplot++Config.cmake.in

```diff
@@ -11,11 +11,24 @@ if (NOT CMAKE_CXX_COMPILER_ID STREQUAL MATPLOT_BUILT_CXX_COMPILER_ID)
 endif()
 
 # Find dependencies
+
 if(NOT ${MATPLOT_BUILT_SHARED})
     include(CMakeFindDependencyMacro)
     list(APPEND CMAKE_MODULE_PATH ${MATPLOT_CONFIG_INSTALL_DIR})
     list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
+
+    # optional dependencies
+    set(MATPLOT_OPTIONAL_DEPENDENCIES JPEG TIFF ZLIB PNG LAPACK BLAS FFTW3 OpenCV)
```

Comment:
The solution has to be more complex than that, which is another drawback of optional dependencies.

[find_dependency](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html) will forward the parameters QUIET and REQUIRED which were passed to the original [find_package()](https://cmake.org/cmake/help/latest/command/find_package.html#command:find_package) call, so the `REQUIRED` parameter below is not really fixing the issue.

This means once the library is installed, there are no more optional dependencies. In the config template:

- The optional dependencies that were not found should have no corresponding find_dependency
- The optional dependencies that were found should have a corresponding find_dependency
- The optional dependencies that were fetched should be installed and added to the targets to export

Of course, another solution is using find_package, but that makes things even more complex because we have to replicate part of the build scripts in the config script.